lib/widgets/chat_list_welcome_tile.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:sloth/extensions/build_context.dart';
+import 'package:sloth/hooks/use_user_metadata.dart';
+import 'package:sloth/src/rust/api/welcomes.dart' show Welcome;
+import 'package:sloth/widgets/wn_animated_avatar.dart';
+
+class ChatListWelcomeTile extends HookWidget {
+  final Welcome welcome;
+  final VoidCallback? onTap;
+
+  const ChatListWelcomeTile({
+    super.key,
+    required this.welcome,
+    this.onTap,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final colors = context.colors;
+
+    final welcomerSnapshot = useUserMetadata(welcome.welcomer);
+    final welcomer = welcomerSnapshot.data;
+    final isLoading = welcomerSnapshot.connectionState == ConnectionState.waiting;
+
+    final welcomerName = welcomer?.displayName ?? welcomer?.name;
+    final hasGroupName = welcome.groupName.isNotEmpty;
+
+    final String avatarDisplayName;
+    final String? avatarPictureUrl;
+    final String title;
+    final String subtitle;
+
+    if (hasGroupName) {
+      avatarDisplayName = welcome.groupName;
+      avatarPictureUrl = null;
+      title = welcome.groupName;
+      subtitle = isLoading
+          ? 'Loading...'
+          : welcomerName != null
+          ? '$welcomerName invited you to a secure chat'
```

Comment:
🦥 Not sure where this kind of logic belongs. In other frameworks I'd have used classes named  "presenters" or "decorators" instead of having this logic directly in the widget

Reply:
But on the other side, some of this texts are very specific of a single widget, like the subtitle.